Tutorial tracking server

lobrien opened the pull request:
## What changes are proposed in this pull request?

New doc: Train methodically with MLflow Tracking

## How is this patch tested?

 Existing unit/integration tests
 New unit/integration tests
[X ] Manual tests (describe details, including test results, below)
Built locally, manual inspection

## Does this PR change the documentation?

 No. You can skip the rest of this section.
[X ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

## Release Notes

Is this a user-facing change?
 No. You can skip the rest of this section.
[ X] Yes. Give a description of this change to be included in the release notes for MLflow users.

New how-to document on MLflow Tracking server and logging. Provides more details and options than Quickstart.

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

 area/artifacts: Artifact stores and artifact logging
 area/build: Build and test infrastructure for MLflow
[ X] area/docs: MLflow documentation pages
 area/examples: Example code
 area/model-registry: Model Registry service, APIs, and the fluent client calls for Model Registry
 area/models: MLmodel format, model serialization/deserialization, flavors
 area/recipes: Recipes, Recipe APIs, Recipe configs, Recipe Templates
 area/projects: MLproject format, project running backends
 area/scoring: MLflow Model server, model deployment tools, Spark UDFs
 area/server-infra: MLflow Tracking server backend
 area/tracking: Tracking Service, tracking client APIs, autologging

Interface

 area/uiux: Front-end, user experience, plotting, JavaScript, JavaScript dev server
 area/docker: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
 area/sqlalchemy: Use of SQLAlchemy in the Tracking Service or Model Registry
 area/windows: Windows support

Language

 language/r: R APIs and clients
 language/java: Java APIs and clients
 language/new: Proposals for new client languages

Integrations

 integrations/azure: Azure and Azure ML integrations
 integrations/sagemaker: SageMaker integrations
 integrations/databricks: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

[ X] rn/documentation - A user-facing documentation change worth mentioning in the release notes

Commit:
Added parallel coordinates image (WIP)

Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

Commit:
Tracking tutorial 1st draft

Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

Commit:
moved sources into subdir

Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

Commit:
Added images

Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

Commit:
2nd draft through artifacts

Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

Commit:
2nd draft

Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

Commit:
New index

Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

Commit:
Split down into separate tutorials per TOC

Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

Commit:
3rd draft

Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

Commit:
Deleted WIP other tutorials

Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

Commit:
Deleted WIP other tutorial docs

Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

Commit:
Removed links to future docs

To be restored after new docs published

Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

Commit:
TOC snafu

Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

Commit:
Formatting and minor edits

Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

mlflow-automation commented:
<!-- documentation preview -->


Documentation preview for 517150bd5474911f0412b8207c9b771b36bfb3d1 will be available [here](https://output.circle-artifacts.com/output/job/ddba27b3-8d4f-40cf-8699-97bc6a7fbe2d/artifacts/0/docs/build/html/index.html) when [this CircleCI job](https://circleci.com/gh/mlflow/mlflow/54681) completes successfully.

<details>
<summary>More info</summary>

- Ignore this comment if this PR does not change the documentation.
- It takes a few minutes for the preview to be available.
- The preview is updated when a new commit is pushed to this PR.
- This comment was created by https://github.com/mlflow/mlflow/actions/runs/5416051081.

</details>

Commit:
Merge branch 'main' into tutorial-tracking-server-2

Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

# Conflicts:
#	docs/source/index.rst
#	docs/source/quickstart.rst

Commit:
index indentation snafu

Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

Commit:
Added `pass` to code snippet to pass lint

Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

Commit:
Cherry pick index from master

Cherry picked index.rst and gateway/index.rst from master for new structure

Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

Commit:
TODO: Add link to tutorial-pyfunc once merged

Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

Commit:
More explicit on using Databricks

Per Jira issue DOC-9763 https://databricks.atlassian.net/jira/software/c/projects/DOC/issues/DOC-9763

Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

Commit:
Removed gateway/index.rst cherrypick

Was breaking the docs build.

TODO: During merge, add gateway/index
Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>
